tests/cnf/ran/ztp/internal/helper/nodedelete.go

```diff
@@ -0,0 +1,182 @@
+package helper
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/assisted"
+	"github.com/openshift-kni/eco-goinfra/pkg/bmh"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/nodes"
+	. "github.com/openshift-kni/eco-gotests/tests/cnf/ran/internal/raninittools"
+	"github.com/openshift-kni/eco-gotests/tests/cnf/ran/ztp/internal/tsparams"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// IsSnoPlusOne checks if the specified cluster has one master and one worker node.
+func IsSnoPlusOne(client *clients.Settings) (bool, error) {
+	masters, err := listNodesByRole(client, "master")
```

Comment:
As of Kubernetes 1.24, the `master` label is no longer applied to nodes:
https://kubernetes.io/blog/2022/04/07/upcoming-changes-in-kubernetes-1-24/#api-removals-deprecations-and-other-changes-for-kubernetes-1-24

`node-role.kubernetes.io/master` changes to `node-role.kubernetes.io/control-plane`.  In the past I've just checked for both `master` and `control-plane` to keep backwards compatibility.

Similar to: https://github.com/openshift-kni/eco-goinfra/pull/265

Reply:
We can potentially do this in a separate PR too to adjust this for the rest of the repo.